clippy_lints/src/lib.rs

```diff
@@ -471,7 +474,22 @@ pub(crate) struct LintInfo {
 pub fn explain(name: &str) {
     let target = format!("clippy::{}", name.to_ascii_uppercase());
     match declared_lints::LINTS.iter().find(|info| info.lint.name == target) {
-        Some(info) => print!("{}", info.explanation),
+        Some(info) => {
+            println!("{}", info.explanation);
+            // Check if the lint has configuration
+            let mdconf = get_configuration_metadata();
+            if let Some(config_vec_positions) = mdconf
+                .iter()
+                .find_all(|cconf| cconf.lints.contains(&info.lint.name_lower()[8..].to_owned()))
+            {
+                // If it has, print it
+                println!("### Configuration for {}:", info.lint.name_lower());
+                for position in config_vec_positions {
+                    let conf = &mdconf[position];
+                    println!("    - {}: {} (default: {})", conf.name, conf.doc, conf.default);
```

Comment:
About the first issue: I'll fix it.
About the second: I have 0 ideas about fixing this. Maybe adding a newline between `### Configuration for ...` and the options?

Reply:
After taking a look at the code again, I don't think I can even fix the first problem.
Each configuration option depends on `mdconf` and `config_vec_positions`. While `mdconf` is independent (taken from `get_configuration_metadata()`), `config_vec_positions` depends on `info`.

I think extrapolating the whole fragment (from line 480 to 491) would abstract so much the process of getting the configuration, that it would difficult future contributors and would make the function more readable, but less understandable overall.

I'll still fix the formatting.

Reply:
Alright, then just keep it as is. Fixing the indention was a nice catch! Thank you, I'll leave you then honer of ordering `@bors` to merge it :D

Reply:
omg thx